### PR TITLE
[GCI] Standardised Core code comments

### LINKED ISSUE #1340: Standardise core code comments
Note: **We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.**

This is a sub-issue of #1046

The comments in the following files are inconsistent:
1. [ImageSequencer.js](https://github.com/publiclab/image-sequencer/blob/src/ImageSequencer.js)
2. [PixelManipulation.js](https://github.com/publiclab/image-sequencer/blob/main/src/modules/_nomodule/PixelManipulation.js)

Rules:
1. Methods must be described using [JSDoc comments](https://devdocs.io/jsdoc/)
2. Misc code comments should be inline unless it is a long sentence.
3. No use of continuous tenses, no pronouns.
4. No redundant comments.
5. Each comment should start with an uppercase letter and end with a full stop.

### PR DESCRIPTION
**Pull request by HarshKhandeparkar**
Fixes #1340  (<=== Replace `0000` with the Issue Number)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

**Comment by SidharthBansal** [comment]
Please check code climate

**Comment by HarshKhandeparkar** [comment]
Code climate errors are caused due to old code. This needs a dedicated pr
to fix. This CC test is ignored in almost every pr.

File: src/ImageSequencer.js [hunk 1]
```diff
@@ -53,22 +70,27 @@ ImageSequencer = function ImageSequencer(options) {
     for (o in sequencer) {
       modules[o] = sequencer[o];
     }
-    sequences = JSON.parse(window.localStorage.getItem('sequences'));
+    sequences = JSON.parse(window.localStorage.getItem('sequences')); // Get saved sequences from localStorage.
     if (!sequences) {
       sequences = {};
-      window.localStorage.setItem('sequences', JSON.stringify(sequences));
+      window.localStorage.setItem('sequences', JSON.stringify(sequences)); // Set the localStorage entry as an empty Object by default.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: Object -> object

File: src/ImageSequencer.js [hunk 2]
```diff
@@ -286,7 +377,7 @@ ImageSequencer = function ImageSequencer(options) {
 
 
   return {
-    //literals and objects
+    // Literals and objects.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: No need of full stop for phrases.
Please do it for all prs

Comment (SidharthBansal) [reply src/ImageSequencer.js]: Use full stop for sentences

File: src/modules/_nomodule/PixelManipulation.js [hunk 3]
```diff
@@ -106,7 +114,7 @@ module.exports = function PixelManipulation(image, options) {
       const inBrowser = (options.inBrowser) ? 1 : 0;
       const test = (process.env.TEST) ? 1 : 0;
       if (options.useWasm) {
-        if (options.inBrowser) {
+        if (options.inBroHawser) {
```
Comment (SidharthBansal) [inline src/modules/_nomodule/PixelManipulation.js]: You did some mistake here

**Comment by HarshKhandeparkar** [comment]
Omg. That is a breaking mistake. Btw, `Object` is a proper noun in JS.

**Comment by HarshKhandeparkar** [comment]
I'll fix the phrases.

File: src/ImageSequencer.js (hunk 1 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]: Object is a proper noun in JS.

File: src/modules/_nomodule/PixelManipulation.js (hunk 3 above)
Comment (HarshKhandeparkar) [reply src/modules/_nomodule/PixelManipulation.js]: fixed

File: src/ImageSequencer.js [hunk 4]
```diff
@@ -182,17 +228,35 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @description Returns the current sequence of steps.
+   * @returns {Object}
+   */
   function getSteps(){
     return this.steps;
   }
 
+  /**
+   * @method setUI
+   * @description To set up a UI for ImageSequencer via different callback methods. Read the docs for more info.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: Which docs? Please add link

File: src/ImageSequencer.js [hunk 5]
```diff
@@ -182,17 +228,35 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @param {Object} UI Object containing UI callback methods. Read the docs for more info.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: Please add doc link

File: src/ImageSequencer.js [hunk 6]
```diff
@@ -222,23 +292,30 @@ ImageSequencer = function ImageSequencer(options) {
     return modulesdata;
   }
 
+  /**
+   * @method loadNewModule
+   * @description Adds a new local module to sequencer. Read the docs for mode info.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: Please add link

File: src/ImageSequencer.js [hunk 7]
```diff
@@ -222,23 +292,30 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @returns {Object}
+   */
   function loadNewModule(name, options) {
 
     if (!options) {
       return this;
 
     } else if (Array.isArray(options)) {
-      // contains the array of module and info
+      // Contains the array of module and info.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: Many extra .
Please correct it

**Comment by SidharthBansal** [comment]
Kindly review prs once yourself too! It will be really helpful to you

File: src/ImageSequencer.js (hunk 7 above)
Comment (HarshKhandeparkar) [inline src/ImageSequencer.js]:
```suggestion
      // Contains the array of module and info
```

File: src/ImageSequencer.js [hunk 8]
```diff
@@ -248,21 +325,35 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @method saveNewModule
+   * @description Saves a new local module to ImageSequencer.
+   * @param {String} name Name of the new module
+   * @param {String} path Path to the new module
+   * @returns {Null}
+   */
   function saveNewModule(name, path) {
     if (options.inBrowser) {
-      // Not for browser context
+      // Not for browser context.
```
Comment (HarshKhandeparkar) [inline src/ImageSequencer.js]:
```suggestion
      // Not for browser context
```

File: src/ImageSequencer.js [hunk 9]
```diff
@@ -182,17 +228,35 @@ ImageSequencer = function ImageSequencer(options) {
 
   }
 
+  /**
+   * @method replaceImage
+   * @description Replaces the current image in the sequencer.
+   * @param {String} selector DOM selector string for the image input.
```
Comment (HarshKhandeparkar) [inline src/ImageSequencer.js]:
```suggestion
   * @param {String} selector DOM selector string for the image input
```

**Comment by HarshKhandeparkar** [comment]
I do review my PRs but I don't seem to find any mistakes. That is what reviewers are meant for I guess. Pointing out others' mistakes (Which I am very good at) :stuck_out_tongue_closed_eyes:

**Comment by SidharthBansal** [comment]
Judge your code as if it was written by someone else

**Comment by HarshKhandeparkar** [comment]
Yeah I know but it usually cannot happen on the same day. Like your mind is
full and you need a bit of a break. I usually can self-review after 24
hours but asking a reviewer to review is better than waiting.

File: src/ImageSequencer.js [hunk 10]
```diff
@@ -14,13 +19,25 @@ ImageSequencer = function ImageSequencer(options) {
     return Object.prototype.toString.call(object).split(' ')[1].slice(0, -1);
   }
 
+  /**
+   * @method log
+   * @description Logs colored messages to the console using ASCII color codes.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 11]
```diff
@@ -14,13 +19,25 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @param {String} color ASCII color code
+   * @param {String} msg Message to be logged to the console.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 12]
```diff
@@ -53,22 +70,27 @@ ImageSequencer = function ImageSequencer(options) {
     for (o in sequencer) {
       modules[o] = sequencer[o];
     }
-    sequences = JSON.parse(window.localStorage.getItem('sequences'));
+    sequences = JSON.parse(window.localStorage.getItem('sequences')); // Get saved sequences from localStorage.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js (hunk 1 above)
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 13]
```diff
@@ -53,22 +70,27 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @method addSteps
+   * @description Adds one of more steps to the sequence.
+   * @return {Object}
+   */
   function addSteps() {
     var this_ = (this.name == 'ImageSequencer') ? this : this.sequencer;
     var args = [];
     var json_q = {};
-    for (var arg in arguments) { args.push(copy(arguments[arg])); }
+    for (var arg in arguments) { args.push(copy(arguments[arg])); } // Get all the module names from the arguments.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 14]
```diff
@@ -77,17 +99,28 @@ ImageSequencer = function ImageSequencer(options) {
...
+  /**
+   * @method removeStep
+   * @description Removes the step at the specified index from the sequence.
+   * @param {Object} ref ImageSequencer instance
+   * @param {Number} index Index of the step to be removed
+   * @returns {Null}
+   */
   function removeStep(ref, index) {
-    //remove the step from images[image].steps and redraw remaining images
+    // Remove the step from images[image].steps and redraw remaining images.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 15]
```diff
@@ -77,17 +99,28 @@ ImageSequencer = function ImageSequencer(options) {
...
       thisStep = ref.steps[index];
       thisStep.UI.onRemove(thisStep.options.step);
       ref.steps.splice(index, 1);
     }
-    //tell the UI a step has been removed
   }
 
+  /**
+   * @method removeSteps
+   * @description Removes one or more steps from the sequence.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 16]
```diff
@@ -103,6 +136,11 @@ ImageSequencer = function ImageSequencer(options) {
     return this;
   }
 
+  /**
+   * @method insertSteps
+   * @description Inserts steps at the specified index.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 17]
```diff
@@ -182,17 +228,35 @@ ImageSequencer = function ImageSequencer(options) {
 
   }
 
+  /**
+   * @method replaceImage
+   * @description Replaces the current image in the sequencer.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 18]
```diff
@@ -182,17 +228,35 @@ ImageSequencer = function ImageSequencer(options) {
...
   function replaceImage(selector, steps, options) {
     options = options || {};
     options.callback = options.callback || function() { };
     return require('./ReplaceImage')(this, selector, steps, options);
   }
 
-  //returns the steps added
+  /**
+   * @method getSteps
+   * @description Returns the current sequence of steps.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 19]
```diff
@@ -201,6 +265,12 @@ ImageSequencer = function ImageSequencer(options) {
     return require('./ExportBin')(dir, this, basic, filename);
   };
 
+  /**
+   * @method modulesInfo
+   * @description Returns information about the given module or all the available modules.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 20]
```diff
@@ -248,6 +325,13 @@ ImageSequencer = function ImageSequencer(options) {
     return this;
   }
 
+  /**
+   * @method saveNewModule
+   * @description Saves a new local module to ImageSequencer.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 21]
```diff
@@ -258,11 +342,18 @@ ImageSequencer = function ImageSequencer(options) {
...
+   * @method saveSequence
+   * @description Saves a sequence on the browser localStorage.
+   * @param {String} name Name for the sequence
+   * @param {String} sequenceString Sequence data as a string
+   * @returns {Null}
+   */
   function saveSequence(name, sequenceString) { // 4. save sequence
     const sequence = str.stringToJSON(sequenceString);
-    // Save the given sequence string as a module
+    // Save the given sequence string as a module.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 22]
```diff
@@ -258,11 +342,18 @@ ImageSequencer = function ImageSequencer(options) {
...
     if (options.inBrowser) {
-      // Inside the browser we save the meta-modules using the Web Storage API
+      // Inside the browser we save the meta-modules using the Web Storage API.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: .

File: src/ImageSequencer.js [hunk 23]
```diff
@@ -276,7 +367,7 @@ ImageSequencer = function ImageSequencer(options) {
   }
 
   function loadModules() {
-    // This function loads the modules and saved sequences
+    // This function loads the modules and saved sequences.
```
Comment (SidharthBansal) [inline src/ImageSequencer.js]: This-> loadModules

File: src/modules/_nomodule/PixelManipulation.js [hunk 24]
```diff
@@ -30,18 +38,18 @@ module.exports = function PixelManipulation(image, options) {
       };
     }
 
-    // iterate through pixels;
+    // Iterate through pixels;
```
Comment (SidharthBansal) [inline src/modules/_nomodule/PixelManipulation.js]: remove ;

File: src/ImageSequencer.js (hunk 10 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Logs colored messages to the console using ASCII color codes
```

File: src/ImageSequencer.js (hunk 11 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @param {String} msg Message to be logged to the console
```

File: src/ImageSequencer.js (hunk 12 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
    sequences = JSON.parse(window.localStorage.getItem('sequences')); // Get saved sequences from localStorage
```

File: src/ImageSequencer.js (hunk 1 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
      window.localStorage.setItem('sequences', JSON.stringify(sequences)); // Set the localStorage entry as an empty Object by default
```

File: src/ImageSequencer.js (hunk 13 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
    for (var arg in arguments) { args.push(copy(arguments[arg])); } // Get all the module names from the arguments
```

File: src/ImageSequencer.js (hunk 14 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
    // Remove the step from images[image].steps and redraw remaining images
```

File: src/ImageSequencer.js (hunk 15 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Removes one or more steps from the sequence
```

File: src/ImageSequencer.js (hunk 16 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Inserts steps at the specified index
```

File: src/ImageSequencer.js (hunk 17 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Replaces the current image in the sequencer
```

File: src/ImageSequencer.js (hunk 18 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Returns the current sequence of steps
```

File: src/ImageSequencer.js (hunk 19 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Returns information about the given module or all the available modules
```

File: src/ImageSequencer.js (hunk 20 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
   * @description Saves a new local module to ImageSequencer
```

File: src/ImageSequencer.js (hunk 21 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
    // Save the given sequence string as a module
```

File: src/ImageSequencer.js (hunk 22 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
      // Inside the browser we save the meta-modules using the Web Storage API
```

File: src/ImageSequencer.js (hunk 23 above)
Comment (HarshKhandeparkar) [reply src/ImageSequencer.js]:
```suggestion
    // loadModules function loads the modules and saved sequences.
```

File: src/modules/_nomodule/PixelManipulation.js (hunk 24 above)
Comment (HarshKhandeparkar) [reply src/modules/_nomodule/PixelManipulation.js]:
```suggestion
    // Iterate through pixels
```

**Comment by jywarren** [comment]
Resolved conflict and will merge when done! Thanks!

**Comment by jywarren** [comment]
Hooray!!

**Pull request merged**